yt_dlp/extractor/bilibili.py

```diff
@@ -502,39 +503,148 @@ def _real_extract(self, url):
             season_info.get('bangumi_title'), season_info.get('evaluate'))
 
 
-class BilibiliChannelIE(InfoExtractor):
-    _VALID_URL = r'https?://space.bilibili\.com/(?P<id>\d+)'
-    _API_URL = "https://api.bilibili.com/x/space/arc/search?mid=%s&pn=%d&jsonp=jsonp"
+class BilibiliSpaceBaseIE(InfoExtractor):
+    def __init__(self, downloader=None):
+        InfoExtractor.__init__(self, downloader)
+        self.page_idx_start = 1
+
+    def _fetch_page(self, playlist_id, page_idx, **kwargs):
+        raise NotImplementedError('This method must be implemented by subclasses')
```

Comment:
I think you can take the query as an argument and avoid using **kwargs

Reply:
the query is URL dependent, I don't think it's a good idea to separate them.
kwargs is just for the convenience of future updates

Reply:
I rewrite whole BilibiliSpaceBaseIE, maybe that is better